gs_quant/api/fred/fredapi.py

```diff
@@ -0,0 +1,305 @@
+            else:
+                raise ValueError('Filter should be a 2 item tuple like (filter_variable, filter_value)')
+
+        sort_order_options = ['asc', 'desc']
+        if sort_order is not None:
+            if sort_order in sort_order_options:
+                url = url + '&sort_order=' + sort_order
+            else:
+                raise ValueError('{} is not in the valid list of sort_order options: {}'.format(sort_order, str(sort_order_options)))
+        
+        # Get first chunk
+        data = self.__fetch_data(url)
+        data = pd.DataFrame(data["seriess"])
+        if limit > 1000:
+            for chunk in range(1, limit // 1000+1):
+                # Fetch data
+                next_data_chunk = self.__fetch_data(url + "&offset={}".format(data.shape[0]))
+                next_data_chunk = pd.DataFrame(next_data_chunk["seriess"])
+                data = data.append(next_data_chunk)
+        return data.head(limit)
+
+    def get_coverage(
+        self,
+        id: str = None,
+        limit: int = None,
+        offset: int = None,
+        fields: List[str] = None,
+        **kwargs
+    ) -> pd.DataFrame:
+        """
+        Get the series covered in the FRED dataset by matching text, release id, or category id. 
+
+        :param id: The dataset id
+        :param limit: The maximum number of results to return (value 0 returns all results)
+        :param offset: The offset
+        :param fields: The fields to return, e.g. assetId
+
+        :Keyword Arguments:
+            * *search_item* (``Union[str, int]``) --
+                Text, category_id or release_id to search 
+            * *search_critera* (``str``) --
+                'text', 'release_id', or 'category_id'
+        :param order_by: Order the results by a criterion ('search_rank', 'field', 'title', 'units', 'frequency',
+            'seasonal_adjustment', 'realtime_start', 'realtime_end', 'last_updated', 'start', 'end',
+            'popularity')
+        :param filter: Filter the results by a criterion ('frequency', 'units', and 'seasonal_adjustment')
+        :return: A DataFrame of the Fred series covered
+
+        **Examples**
+
+        >>> from gs_quant.api.fred import FredDataApi
+        >>> 
+        >>> API_KEY = <YOUR API KEY>
+        >>> fredAPI = FredDataApi(api_key = API_KEY)
+        >>> fred_data = Dataset("FRED", fredAPI)
+
+        >>> fred_data.get_coverage(search_item="GDP", search_criteria="text")
+        >>> fred_data.get_coverage(search_item=151, search_criteria="release_id")
+        >>> fred_data.get_coverage(search_item=33058, search_criteria="category_id")
+        """
+        
+        assert("search_criteria" in kwargs)
+        assert("search_item" in kwargs)
+        if limit is None:
+            limit = 1000
+        search_critera = kwargs["search_criteria"]
+        search_item = kwargs["search_item"]
+
+        if search_critera == 'text':
```

Comment:
The following is common to all clauses and be written once at the end:

coverage = self.__get_search_results(url=url, limit=limit)
if coverage is None:
    raise ValueError('No series exists for text: ' + str(search_item))